src/Xrefcheck/Scanners/Markdown.hs

```diff
@@ -176,8 +180,34 @@ nodeExtractInfo input@(Node _ _ nSubs) = do
         _ -> return mempty
 
     copyPaste :: Node -> m FileInfoDiff
+    copyPaste (Node _ (LIST _) nodes) = do
+      case items of
+        top : rest | urlIsASubsequence top -> do
+          let bad = filter (not . urlIsASubsequence) rest
```

Comment:
According to the [acceptance criteria](https://github.com/serokell/xrefcheck/issues/64#issue-866106373), we want to report copy-paste if `there are two links [T1](L1) and [T2](L1) in a file, and T1 is substring of L1 modulo casing and all the non-alphanum characters, while T2 is not substring of L1 modulo the same things;`

However, AFAIU here you check that the first item satisfy the law -- `T1` is a subsequence of `L1` -- and report all other links from the list that doesn't satisfy as invalid, so `[Tn](Ln)` would also be reported if `Tn` isn't a subsequence of `Ln`, e.g.:
```
- [Foo Bar](foo-bar) e
- [Foo Qux](foo-qux) e
- [Foo Kek](foo-kek) e
- [Just text](file) e
```
produces
```
  ➥  In file tests/markdowns/without-annotations/copy/copy-paste.md
     bad reference (local) at src:18:4-12:
       - text: ""
       - link:
       - anchor: -

     ⛂  Possibly incorrect copy-paste in list with references
        the url is file
           but the text is Just text
```
I am not sure if we need to consider such list items as bad copy-paste, I believe it would be better to don't take them into account at all and report only those which strictly satisfy the law from acceptance criteria -- `there are two links [T1](L1) and [T2](L1) in a file, and T1 is substring of L1 modulo casing and all the non-alphanum characters, while T2 is not substring of L1 modulo the same things;`